File: Two Pointers双指针(对撞指针)/75. Sort Colors.py
class Solution:
    def sortColors(self, nums):
        zero = -1
        i = 0 
        two = len(nums)
        while i < two:
            if nums[i] == 1:
                i += 1
            elif nums[i] == 2:
                two -= 1
                nums[two], nums[i] = nums[i], nums[two]
            else:
                zero += 1
                nums[zero], nums[i] = nums[i], nums[zero]
                i += 1

File: 75. Sort Colors.cpp
class Solution {
public:
    void sortColors(vector<int>& nums) {
        int i = -1,j = -1,k = -1;
        int m;
        for(m = 0; m < nums.size();m++){
            if(nums[m] == 0){
                nums[++k] = 2;
                nums[++j] = 1;
                nums[++i] = 0;
            }
            else if(nums[m] == 1){
                nums[++k] = 2;
                nums[++j] = 1;
            }
            else {
                nums[++k] = 2;
            }
        }
    }
};
